src/main/start-main-application/lens-window/application-window/create-electron-window.injectable.ts

```diff
@@ -79,6 +87,14 @@ const createElectronWindowInjectable = getInjectable({
         },
       });
 
+      if (isLinux) {
+        const iconFileName = getAbsolutePath(lensResourcesDir, "../open-lens.png");
```

Comment:
This files doesn't exist in the repo, where does it get generated?

Reply:
good question, I thinks it's generated at build time from a svg ?

![image](https://user-images.githubusercontent.com/117580/197050410-d560a2cf-db7f-4c49-9344-3d048fc35c47.png)

Reply:
Do you just run `make build`?

Reply:
I did a `make build`, then run the generated dist AppImage

Reply:
Ah I think I understand. You are inside the mount that the AppImage produces.

Reply:
yes, I'm unable to find a "icon asset" injectable :/

Reply:
`${packager.executableName}` in `./node_modules/app-builder-lib/out/targets/fpm.js`, still digging

Reply:
`const iconFileName = getAbsolutePath(lensResourcesDir, "../open-lens.png");` should be fine.

The only thing that needs to be changed is the file name itself. To get that please do the following:

1. In `src/common/vars/application-information.injectable.ts` change the `ApplicationInformation` type to include the `"name"` field.
2. Add a `name` destructure to fix the type error within that file.
3. Use `applicationInformationInjectable` in `src/main/start-main-application/lens-window/application-window/create-electron-window.injectable.ts` to produce the file name, which should be `../${applicationInformation.name}.png`

Reply:
I have two paths to test and several options:
 - first is to set one icon location based on `process.env["APPIMAGE"] == null`
 - second is to test two paths but I end up with nested promise which is quite inelegant

Any suggestions ?

Reply:
Or I go to `di.inject(fsInjectable).existsSync` ?

Reply:
```node
        let iconFileName = [
          getAbsolutePath(lensResourcesDir, `../${applicationInformation.name}.png`),
          `/usr/share/icons/hicolor/512x512/apps/${applicationInformation.name}.png`,
        ].find(di.inject(fsInjectable).existsSync);

        if (iconFileName != undefined) {
          try {
            browserWindow.setIcon(iconFileName);
          } catch (err) {
            logger.warn(`Error while setting window icon ${err}`);
          }
        } else {
          logger.warn(`No suitable icon found for task bar.`);
        }
```
 @Nokel81 what about this one ?

Reply:
That looks file, I would just add some prefix to the log calls, something like `[ELECTRON-WINDOW]:`

Reply:
Done :)